bentoml/repository/__init__.py

```diff
@@ -147,11 +153,31 @@ def add(self, bento_name, bento_version):
 
     def get(self, bento_name, bento_version):
         # Return s3 path containing uploaded Bento files
-        raise NotImplementedError
+
+        object_name = self._get_object_name(bento_name, bento_version)
+
+        try:
+            response = self.s3_client.generate_presigned_url(
+                'get_object',
+                Params={'Bucket': self.bucket, 'Key': object_name},
+                ExpiresIn=self._expiration,
+            )
+        except Exception as e:
+            raise BentoMLRepositoryException(
+                "Not able to get pre-signed URL on S3. Error: {}".format(e)
+            )
+        return response
 
     def dangerously_delete(self, bento_name, bento_version):
         # Remove s3 path containing related Bento files
-        raise NotImplementedError
+
+        object_name = self._get_object_name(bento_name, bento_version)
+
+        response = self.s3_client.delete_object(Bucket=self.bucket, Key=object_name)
```

Comment:
does this raise a ClientError when object no exist? can we improve the error message we show to users in that case?

Reply:
different people think differently about this, I want to hear from you on this.
when trying to delete an object, if the object is not there, should we return success or failure?
what if a user tried to delete the same object twice?
(should we make endpoints idempotently safe? (even on what is idempotency, there are different opinions...))

Reply:
I think it should return "deleted" when the object is not found, similar behavior to the s3 delete_object API.

My concern here was - we should try to catch ClientError around delete_object call, in case the request timeout or anything like that happened. And in that case, we want to know it failed to delete the object on the cloud platform.

Reply:
Got it. I'll wrap a try-catch then.